controller/apierror/api.go

```diff
@@ -34,10 +38,71 @@ func (e ApiError) Error() string {
 	if e.Cause != nil && e.AppendCause {
 		s = s + ": " + e.Cause.Error()
 	}
-
 	return s
 }
 
+func (e ApiError) ToRestModel(requestId string) *rest_model.APIError {
+
+	ret := &rest_model.APIError{
+		Args:      nil,
+		Code:      e.Code,
+		Message:   e.Message,
+		RequestID: requestId,
+	}
+
+	if e.Cause != nil {
+		if causeApiError, ok := e.Cause.(*ApiError); ok {
+			//standard apierror
+			ret.Cause = &rest_model.APIErrorCause{
+				APIError: *causeApiError.ToRestModel(requestId),
+			}
+		} else if causeJsonSchemaError, ok := e.Cause.(*schema.ValidationErrors); ok {
+			//only possible from config type JSON schema validation
+			ret.Cause = &rest_model.APIErrorCause{
+				APIFieldError: rest_model.APIFieldError{
+					Field:  causeJsonSchemaError.Errors[0].Field,
+					Reason: causeJsonSchemaError.Errors[0].Error(),
+					Value:  fmt.Sprintf("%v", causeJsonSchemaError.Errors[0].Value),
```

Comment:
Errors[0] is guaranteed to exist? I found other places in here where [0] existed. i ask that you just reaffirm that they don't need boundary checking

Reply:
It is, it is a composite error.